### ISSUE TITLE
Unable to Reproduce Metric Results from Paper and README

### ISSUE THREAD

**Issue by JiangJixiang** [synsin0/StreamingFlow#2]
Thanks for your great work, but I got some problems to reproduce Metric Results from Paper and README.

Environment

- PyTorch == 1.10.1
- PyTorch-Lightning == 1.6.*

When running the evaluation command:

```bash
python evaluate.py --checkpoint ./ckpts/streamingflow_lc_ode_variable.ckpt
```

I obtain unexpectedly high IoU:

```yaml
vehicle_iou: 0.9746925830841064
vehicle_pq: 0.2245146781206131
vehicle_sq: 0.7242259979248047
vehicle_rq: 0.31000638008117676
```

These results differ significantly from both:

- The README metrics (IoU: 53.7, VPQ: 50.7)
- All metrics reported in the original paper

After modifying the [line](https://github.com/synsin0/StreamingFlow/blob/main/streamingflow/metrics.py#L38) to:

```python
tps, fps, _, fns, sups = multiclass_stat_scores(prediction, target, self.n_classes, average=None).t()
```

The results became:

```yaml
vehicle_iou: 0.08851978927850723
vehicle_pq: 0.08614833652973175
vehicle_sq: 0.6942065954208374
vehicle_rq: 0.1240961030125618
```

How can I fix the following problems:

- Initial results show suspiciously high IoU (0.9747) but low PQ
- After modification, IoU becomes implausibly low (0.0885)
- Why the two vehicle_sqs are different to each other.

**Comment by synsin0** [comment]
The results are abnormal. Please adjust pytorch_lightning version to 1.2.5 and test my provided checkpoint for the IoU and PQ results. I am afraid that the version of lightning and torch_metrics makes some difference.

**Comment by synsin0** [comment]
Please check three key points. The version of pytorch_lightning==1.2.5, torchmetrics==0.6.0, mmdet3d should be developed from source (python setup.py develop), It should be mmdet3d-bevfusion in the root directory, this is because mmdet3d from mit-han-lab has a rotated feature map in the sparse_encoder, but this operation is not in the official mmdet3d.

**Comment by JiangJixiang** [comment]
Thank you for your answer.

After `pip install pytorch_lightning==1.2.5 torchmetrics==0.6.0`, and `python setup.py develop`, and `python evaluate.py --checkpoint ./ckpts/streamingflow_lc_ode_variable.ckpt`, I got:

```yaml
vehicle_iou : 0.536292314529419
vehicle_pq : 0.5063307285308838
vehicle_sq : 0.7541001439094543
vehicle_rq : 0.6714370250701904
```

The results are close to the metrics listed in the README (IoU: 53.7, VPQ: 50.7), but they differ significantly from the metrics reported in the StreamingFlow paper.

Does the evaluation in the paper include additional classes beyond vehicles? Could this explain the discrepancy?

**Comment by synsin0** [comment]
The open-sourced model is set with image size 224x480 at 2Hz and LiDAR VoxelNet at 5Hz to demonstrate its ability to handle continuous prediction with asynchronous sensors. This model is trained with 4 A6000 from scratch in this codebase. The models in the paper are under different image sizes, image and lidar backbones, frequencies and have some pretraining techniques.

**Comment by JiangJixiang** [comment]
Are the metrics computed just based on the Vehicle class in the paper as the `evaluate.py` does?

**Comment by synsin0** [comment]
Yes, the metrics are the same. The model gets IoU=53.7 and PQ=50.7 under the metrics proposed by FIERY.

**Comment by JiangJixiang** [comment]
Thanks a lot! I’m going to experiment with some ideas based on streamingflow—your answers have been very helpful.
